### ISSUE TITLE
[Documentation] Specs vs XML differences and questions LevelControlCluster

### ISSUE THREAD

**Issue by Apollon77** [project-chip/connectedhomeip#25611]
### Documentation issues

I discovered the following differences:
* Attribute CurrentLevel has a default value "null" according to specs, but "0" in XML. Which is right?
* Attribute minLevel Default value is formally dynamic based on feature support and might be not correct with being 0 in the XML
* Attribute onLevel has a default value of "Null" according to specs, XML do not specify any default value
* Attribute CurrentFrequency is in the end defined my min and max attributes. Why the default value is 0 then (in specs and XML)? (comparable for maxFrequency)

**Comment by Apollon77** [comment]
> There's no way to specify the dynamic thing in the XML.

Yes thats right and also ok ... Interestingly some clusters then use the "maximum possible border min/max values" and some others simply do not specify min/max at all in this case ... ideally this shpuld also be consistent ... so using the "most outside border values" everywhere or not at all

**Comment by bzbarsky-apple** [comment]
> ideally this shpuld also be consistent ... so using the "most outside border values" everywhere or not at all

In practice, min/max in the XML only matters for writable attributes, in terms of the observable behavior.  And even then more constraints can be done manually...  

> With the info I got from @turon that the XML is "the only truth"

That's the goal, yes.  We're pretty far from it, as you have noticed.....

**Comment by Apollon77** [comment]
We planned to auto generate the Cluser code based on the XMLs for the TypeScript library, so we would also use it for "generating the types", shich means that also the developer that implements a device itself could make use of it because we make sure he uses correct values at all (not only when he sends a write request) :-)

But ok, knowing that we might need a more "use generated as basis and enhance" approach ...

> We're pretty far from it, as you have noticed.....

Glad I could help to come closer :-)

**Comment by andy31415** [comment]
Removed cert blocker during triage: we expect this needs fixing, however XML differences likely do not affect SDK ability to be certified (at least regarding these issues, given that certification  tests still pass).
